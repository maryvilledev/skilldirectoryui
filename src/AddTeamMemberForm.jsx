import React, { PropTypes } from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import { Col, ControlLabel, Button,
   Form, FormControl, FormGroup } from 'react-bootstrap';

class AddTeamMemberForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      teamMemberName: '',
      teamMemberTitle: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onNameChange = this.onNameChange.bind(this);
    this.onTitleChange = this.onTitleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onChange(key) {
    // Return a curried function that sets a value to the key in this.state
    return (value) => {
      this.setState({
        [key]: value,
      });
    };
  }
  onNameChange(ev) {
    this.onChange('teamMemberName')(ev.target.value);
  }
  onTitleChange(ev) {
    this.onChange('teamMemberTitle')(ev.target.value);
  }
  onSubmit(ev) {
    ev.preventDefault();
    const name = this.state.teamMemberName;
    if (!name) {
      alert('Please enter a name');
      return;
    }
    const title = this.state.teamMemberTitle;
    if (!title) {
      alert('Please enter a title');
      return;
    }
    this.props.onSubmit(name, title);
  }
  render() {
    return (
      <div>
        <h1>Add a Team Member</h1>
        <Form horizontal onSubmit={this.onSubmit}>
          <FormGroup controlId="teamMemberName">
            <Col componentClass={ControlLabel} sm={2}>
              Name:
            </Col>
            <Col sm={10}>
              <FormControl
                name="teamMemberName"
                componentClass="input"
                onChange={this.onNameChange}
              />
            </Col>
          </FormGroup>

          <FormGroup controlId="teamMemberTitle">
            <Col componentClass={ControlLabel} sm={2}>
              Title:
            </Col>
            <Col sm={10}>
              <FormControl
                name="teamMemberTitle"
                componentClass="input"
                onChange={this.onTitleChange}
              />
            </Col>
          </FormGroup>

          <FormGroup>
            <Col smOffset={2} sm={2}>
              <Button type="submit" bsStyle="primary">
                Submit
              </Button>
            </Col>
            <Col smOffset={2}>
              <Button onClick={this.props.closeModal} bsStyle="info">
                Cancel
              </Button>
            </Col>
          </FormGroup>
        </Form>
      </div>
    );
  }
}

AddTeamMemberForm.propTypes = {
  closeModal: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default AddTeamMemberForm;
