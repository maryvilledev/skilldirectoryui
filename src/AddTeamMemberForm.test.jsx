import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';

import AddTeamMemberForm from './AddTeamMemberForm';

describe('<AddTeamMemberForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddTeamMemberForm />, div);
  });

  it('matches the stored snapshot', () => {
    const wrapper = shallow(<AddTeamMemberForm />);
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('should not attempt to submit if a field is empty', () => {
    const handleSubmit = jest.fn();
    const form = mount(<AddTeamMemberForm onSubmit={handleSubmit} />);
    const formElem = form.find('form').first();
    formElem.simulate('submit');
    expect(handleSubmit).not.toHaveBeenCalled();
  });
  it('will call the callback when the fields are populated', () => {
    const handleSubmit = jest.fn();
    const form = mount(<AddTeamMemberForm onSubmit={handleSubmit} />);
    form.setState({ teamMemberName: 'foo', teamMemberTitle: 'bar' });
    const formElem = form.find('form').first();
    formElem.simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
