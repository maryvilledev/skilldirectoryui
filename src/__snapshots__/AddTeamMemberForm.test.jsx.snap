exports[`<AddTeamMemberForm /> matches the stored snapshot 1`] = `
<div>
  <h1>
    Add a Team Member
  </h1>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    onSubmit={[Function]}>
    <FormGroup
      bsClass="form-group"
      controlId="teamMemberName">
      <Col
        bsClass="col"
        componentClass={[Function]}
        sm={2}>
        Name:
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="input"
          name="teamMemberName"
          onChange={[Function]} />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="teamMemberTitle">
      <Col
        bsClass="col"
        componentClass={[Function]}
        sm={2}>
        Title:
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="input"
          name="teamMemberTitle"
          onChange={[Function]} />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
        smOffset={2}>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          type="submit">
          Submit
        </Button>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        smOffset={2}>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="info"
          disabled={false}
          onClick={undefined}>
          Cancel
        </Button>
      </Col>
    </FormGroup>
  </Form>
</div>
`;
